src/llama_stack/providers/remote/prompts/mlflow/mlflow.py

```diff
@@ -0,0 +1,565 @@
+                # In this case, we need to list all and filter
+                prompts = mlflow.genai.search_prompts()
+        except Exception as e:
+            logger.error(f"Failed to search prompts in MLflow: {e}")
+            return ListPromptsResponse(data=[])
+
+        llama_prompts = []
+        for mlflow_prompt in prompts:
+            # Filter by name prefix if not using metadata
+            if not self.config.use_metadata_tags:
+                if not mlflow_prompt.name.startswith(self.mapper.MLFLOW_NAME_PREFIX):
+                    continue
+
+            try:
+                # Convert MLflow name to Llama Stack ID
+                prompt_id = self.mapper.to_llama_id(mlflow_prompt.name)
+
+                # Get default version
+                llama_prompt = await self.get_prompt(prompt_id)
+                llama_prompts.append(llama_prompt)
+            except (ValueError, Exception) as e:
+                # Skip prompts that can't be converted or retrieved
+                logger.warning(f"Skipping prompt {mlflow_prompt.name}: {e}")
+                continue
+
+        # Sort by prompt_id
+        llama_prompts.sort(key=lambda p: p.prompt_id, reverse=True)
+
+        return ListPromptsResponse(data=llama_prompts)
+
+    async def list_prompt_versions(self, prompt_id: str) -> ListPromptsResponse:
+        """List all versions of a specific prompt.
+
+        Args:
+            prompt_id: Llama Stack prompt ID
+
+        Returns:
+            ListPromptsResponse with all versions of the prompt
+
+        Raises:
+            ValueError: If prompt not found
+        """
+        # MLflow doesn't have a direct "list versions" API for prompts
+        # We need to iterate and try to load each version
+        versions = []
+        version_num = 1
+        max_attempts = 100  # Safety limit
+
+        while version_num <= max_attempts:
+            try:
+                prompt = await self.get_prompt(prompt_id, version_num)
+                versions.append(prompt)
+                version_num += 1
+            except ValueError:
+                # No more versions
+                break
+            except Exception as e:
+                logger.warning(f"Error loading version {version_num} of {prompt_id}: {e}")
+                break
+
+        if not versions:
+            raise ValueError(f"Prompt {prompt_id} not found")
+
+        # Sort by version number
+        versions.sort(key=lambda p: p.version)
+
+        return ListPromptsResponse(data=versions)
+
+    async def set_default_version(self, prompt_id: str, version: int) -> Prompt:
+        """Set default version using MLflow alias.
+
+        Args:
+            prompt_id: Llama Stack prompt ID
+            version: Version number to set as default
+
+        Returns:
+            Prompt resource with is_default=True
+
+        Raises:
+            ValueError: If version not found
+        """
+        import mlflow
```

Comment:
is there a reason to lazily load mlflow at the start of each method rather than at the top of the file?

Reply:
good catch, I created self-contained functions during dev and left them like that when converting them to methods. I'll fix this.